### PR TITLE
Project remove properties

### PR DESCRIPTION
**Pull request by robinschmid**
The properties in the project are annoying. Currently, it is not safe how we add new elements etc. due to the bindings to JavaFX components. 

I changed the structure back to simple ObservableLists and change listeners, where needed. Other components just use the copies of the current lists.

### CODE REVIEW

File: src/main/java/io/github/mzmine/datamodel/features/ModularFeatureList.java [hunk 1]
```diff
@@ -149,9 +150,9 @@ public String setName(String name) {
     final MZmineProject project = MZmineCore.getProjectManager().getCurrentProject();
 
     if (project != null) {
-      synchronized (project.getFeatureLists()) {
+      synchronized (project.getCurrentFeatureLists()) {
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/datamodel/features/ModularFeatureList.java:153]: no point in synchronizing on an object created on method call, or?

File: src/main/java/io/github/mzmine/gui/MZmineGUI.java [hunk 2]
```diff
@@ -178,15 +179,19 @@ public static void activateProject(MZmineProject project) {
...
-        GroupableListView<FeatureList> featureListsList = mainWindowController
-            .getFeatureListsList();
-        featureListsList.setValues(project.featureListsProperty().getValue());
+        GroupableListView<FeatureList> featureListsList = mainWindowController.getFeatureListsList();
+        featureListsList.setValues(project.getObservableFeatureLists());
 
         var libraryList = mainWindowController.getSpectralLibraryList();
-        libraryList.setItems(project.spectralLibrariesProperty().getValue());
+        final ObservableList<SpectralLibrary> libs = project.getObservableSpectralLibraries();
+        final var fxLibs = FXCollections.observableArrayList(new ArrayList<>(libs));
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/gui/MZmineGUI.java:189]: why are you wrapping an observable list?

File: src/main/java/io/github/mzmine/modules/dataprocessing/featdet_chromatogramdeconvolution/FeatureResolverSetupDialog.java [hunk 3]
```diff
@@ -103,7 +103,7 @@ public FeatureResolverSetupDialog(boolean valueCheckRequired, ParameterSet param
     previewChartBadFeature.setRangeAxisNumberFormatOverride(intensityFormat);
 
     ObservableList<ModularFeatureList> flists = (ObservableList<ModularFeatureList>) (ObservableList<? extends FeatureList>) MZmineCore.getProjectManager()
-        .getCurrentProject().getFeatureLists();
+        .getCurrentProject().getCurrentFeatureLists();
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/dataprocessing/featdet_chromatogramdeconvolution/FeatureResolverSetupDialog.java:106]: 1. you are casting a normal array list to observable here. 
2. Why not use the observable directly?

File: src/main/java/io/github/mzmine/modules/dataprocessing/featdet_masscalibration/MassCalibrationSetupDialog.java [hunk 4]
```diff
@@ -103,8 +113,8 @@ public MassCalibrationSetupDialog(boolean valueCheckRequired, ParameterSet param
     pnlDataFile = new FlowPane();
     pnlDataFile.getChildren().add(new Label("Data file "));
 
-    comboDataFileName = new ComboBox<RawDataFile>(
-        MZmineCore.getProjectManager().getCurrentProject().getRawDataFiles());
+    comboDataFileName = new ComboBox<>(FXCollections.observableList(
+        MZmineCore.getProjectManager().getCurrentProject().getCurrentRawDataFiles()));
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/dataprocessing/featdet_masscalibration/MassCalibrationSetupDialog.java:117]: why not use the observable lists?

File: src/main/java/io/github/mzmine/modules/dataprocessing/featdet_mobilogram_summing/MobilogramBinningSetupDialog.java [hunk 5]
```diff
@@ -70,24 +70,22 @@ public MobilogramBinningSetupDialog(boolean valueCheckRequired,
...
+        MZmineCore.getConfiguration().getMobilityFormat());
     previewChart.setMinHeight(400);
     processedRenderer = new ColoredXYShapeRenderer();
 
     previewChart.setRangeAxisNumberFormatOverride(intensityFormat);
-    ObservableList<ModularFeatureList> flists = (ObservableList<ModularFeatureList>)
-        (ObservableList<? extends FeatureList>) MZmineCore.getProjectManager().getCurrentProject()
-            .getFeatureLists();
+    ObservableList<ModularFeatureList> flists = (ObservableList<ModularFeatureList>) (ObservableList<? extends FeatureList>) MZmineCore.getProjectManager()
+        .getCurrentProject().getCurrentFeatureLists();
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/dataprocessing/featdet_mobilogram_summing/MobilogramBinningSetupDialog.java:80]: again, casting normal lsit to observable

File: src/main/java/io/github/mzmine/modules/dataprocessing/featdet_smoothing/SmoothingSetupDialog.java [hunk 6]
```diff
@@ -76,18 +76,16 @@ public SmoothingSetupDialog(boolean valueCheckRequired,
     previewDimension = SmoothingDimension.RETENTION_TIME;
     previewChart.setDomainAxisNumberFormatOverride(rtFormat);
     previewChart.setRangeAxisNumberFormatOverride(intensityFormat);
-    ObservableList<ModularFeatureList> flists = (ObservableList<ModularFeatureList>)
-        (ObservableList<? extends FeatureList>) MZmineCore.getProjectManager().getCurrentProject()
-            .getFeatureLists();
+    ObservableList<ModularFeatureList> flists = (ObservableList<ModularFeatureList>) (ObservableList<? extends FeatureList>) MZmineCore.getProjectManager()
+        .getCurrentProject().getCurrentFeatureLists();
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/dataprocessing/featdet_smoothing/SmoothingSetupDialog.java:80]: again

File: src/main/java/io/github/mzmine/modules/io/projectsave/ProjectSavingTask.java [hunk 7]
```diff
@@ -349,7 +349,7 @@ private void saveRawDataFiles(ZipOutputStream zipStream)
   private void savePeakLists(ZipOutputStream zipStream)
       throws IOException, TransformerConfigurationException, SAXException {
 
-    for (FeatureList featureList : savedProject.getFeatureLists()) {
+    for (FeatureList featureList : savedProject.getCurrentFeatureLists()) {
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/io/projectsave/ProjectSavingTask.java:352]: will this create a new list on every iteration?

File: src/main/java/io/github/mzmine/modules/visualization/spectra/simplespectra/SpectraBottomPanel.java [hunk 8]
```diff
@@ -85,8 +86,8 @@
 
     Label featureListLabel = new Label("Feature list: ");
 
-    peakListSelector = new ComboBox<FeatureList>(
-        MZmineCore.getProjectManager().getCurrentProject().getFeatureLists());
+    peakListSelector = new ComboBox<>(FXCollections.observableList(
+        MZmineCore.getProjectManager().getCurrentProject().getCurrentFeatureLists()));
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/visualization/spectra/simplespectra/SpectraBottomPanel.java:90]: why not use the observable list?

File: src/main/java/io/github/mzmine/project/impl/RawDataFileImpl.java [hunk 9]
```diff
@@ -472,9 +473,9 @@ public String setName(@NotNull String name) {
     final MZmineProject project = MZmineCore.getProjectManager().getCurrentProject();
 
     if (project != null) {
-      synchronized (project.getRawDataFiles()) {
+      synchronized (project.getCurrentRawDataFiles()) {
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/project/impl/RawDataFileImpl.java:476]: same as with feature list

**Review by SteffenHeu — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/mzmine/modules/io/projectsave/ProjectSavingTask.java (hunk 7 above)
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/io/projectsave/ProjectSavingTask.java:352]: I think it wont. it should be the same like getCurrentFeatureLists().forEach

**Comment by robinschmid** [comment]
Together with @SteffenHeu

**Pull request merged**